smtlib-backends-process/tests/Examples.hs

```diff
@@ -0,0 +1,85 @@
+{-# LANGUAGE OverloadedStrings #-}
+
+module Examples (examples) where
+
+import qualified Data.ByteString.Lazy.Char8 as LBS
+import Data.Default (def)
+import SMTLIB.Backends (command, command_, initSolver)
+import qualified SMTLIB.Backends.Process as Process
+import System.Exit (ExitCode (ExitSuccess))
+import System.IO (BufferMode (LineBuffering), hSetBuffering)
+import System.Process.Typed (getStdin)
+import Test.Tasty
+import Test.Tasty.HUnit
+
+-- | The examples for the 'Process' backend (running solvers as external
+-- processes).
+examples :: [TestTree]
+examples =
+  [ testCase "basic use" basicUse,
+    testCase "setting options" setOptions,
+    testCase "exiting manually" manualExit
+  ]
+
+-- | Basic use of the 'Process' backend.
+basicUse :: IO ()
+basicUse =
+  -- 'Process.with' runs a computation using the 'Process' backend
+  Process.with
+    -- the configuration type 'Process.Config' is an instance of the 'Default' class
+    -- we can thus use a default configuration for the backend with the 'def' method
+    -- this default configuration uses Z3 as an external process and disables logging
+    def
+    $ \handle -> do
+      -- first, we make the process handle into an actual backend
+      let backend = Process.toBackend handle
+      -- then, we create a solver out of the backend
+      -- we enable queuing (it's faster !)
+      solver <- initSolver backend True
+      -- we send a basic command to the solver and ignore the response
+      -- we can write the command as a simple string because we have enabled the
+      -- OverloadedStrings pragma
+      _ <- command solver "(get-info :name)"
+      return ()
+
+-- | An example of how to set options for the 'Process' backend.
```

Comment:
I think it makes sense to keep "default" because the previous examples uses the default settings so the reader naturally wonders how to actually change these default settings.